engine.py

```diff
@@ -52,24 +52,82 @@ def get_pawn_moves(self, r, c, moves):
 			return parameter(s):
 			None
 		"""
+		
+		def pawn_attack(pawn_type):
+			"""
+			The pawn attack in diagonal position against opponent piece
+	
+			"""
+			if pawn_type == 'l':
+				if c == 0 and self.board[r-1][c+1][1] == 'd':
+					return moves.append(Move((r, c), (r-1,c+1), self.board))			
+				elif c == 7 and self.board[r-1][c-1][1] == 'd':
+					return moves.append(Move((r, c), (r-1,c-1), self.board))
+				elif c not in [0,7]:
+					for i in range(-1, 2, 2):  # loops twice so j == -1 or 1 
+						if self.board[r - 1][c + i][1] == "d":  # if there's a dark piece is diagonal to it
+							return moves.append(Move((r, c), (r-1, c+i), self.board))  # create  move object and append to list
+					
+			elif pawn_type == 'd':
+				if c == 0 and self.board[r+1][c+1][1] == 'l':
+					return moves.append(Move((r, c), (r+1, c+1), self.board))
+				elif c == 7 and self.board[r+1][c-1][1] == 'l':
+					return moves.append(Move((r, c), (r+1, c-1), self.board))
+				elif c not in [0,7]:
+					for i in range(-1, 2, 2):  # loops twice so j == -1 or 1 
+						if self.board[r+1][c+i][1] == 'l':  # if there's a dark piece is diagonal to it
+							return moves.append(Move((r, c), (r+1, c+i), self.board))  # create  move object and append to list			
+			
 
 		## FIX
 		if self.light_to_move: # if it's light's turn to move
 			
 			for i in range(len(self.board)):
 				for j in range(len(self.board[i])):
```

Comment:
It appears i and j are never used inside the loop! this nested loop creates the same move object and appends to list 64 times! (8 * 8) which isn't neccessary